### PR TITLE
Chiu Chi Kuan ip

### PR DESCRIPTION
**Pull request by zlimez**
# Add **CRUD** functionalities 
Zlimez, a task management app, helps you track the countless number of tasks you have to do while encouraging you to still make good use of your memory so that you do not become a goldfish someday. 
> "You can’t trust other people. If it’s important, you have to do it yourself" - _Neil Gaiman_

Download from [here](https://github.com/zlimez/ip/releases/tag/A-Jar) to play around and see for yourself the temperament of Zlimez and what he can do for you 😡.

Current features include:
- Add tasks
- Mark tasks
- Delete tasks
- List tasks _(All or those on a specific date)_
- Find tasks

Event types currently supported include:
1. General todo tasks
2. Events
3. Tasks with deadline

Long term product goals:
- [x] Task Management
- [ ] GUI (coming soon)
- [ ] Conversational

A snapshot of Zlimez's response, assuming you understand Java:
```
  public void bye() {
      System.out.println("\tBye. zzz FINALLY~~" + " " + EMOJI);
      reader.close();
  }
```

### CODE REVIEW

File: src/main/java/chatbot/storage/Storage.java [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+
+/**
+ * The class is responsible for populating the chatbot with possible data
+ * from previous sessions and saving the data of the current session just
+ * before it ends.
+ */
+public class Storage {
+    private static final String DIVIDER = " \\| ";
+    private static final String ROOT = System.getProperty("user.dir");
+    private static final Path DATA_PATH = Paths.get(ROOT, "data", "data.txt");
```
Comment (highorbit25) [inline src/main/java/chatbot/storage/Storage.java]: Good job, defining these constants improved code readability greatly.

File: text-ui-test/input.txt [hunk 2]
```diff
@@ -0,0 +1,19 @@
+sleep
+todo sleep
+list
+mark 2
+mark 1
+unmark 1
+deadline kiss you /at 2022-09-15
+deadline kiss you /by 2022-09-15
```
Comment (highorbit25) [inline text-ui-test/input.txt:8]: HAHA 💯

File: src/main/java/chatbot/tasks/TaskList.java [hunk 3]
```diff
@@ -0,0 +1,164 @@
+package chatbot.tasks;
+
+import chatbot.exceptions.DukeException;
+
+import java.time.LocalDate;
+import java.util.*;
```
Comment (highorbit25) [inline src/main/java/chatbot/tasks/TaskList.java]: Perhaps can try to avoid wildcard imports

File: src/main/java/chatbot/ui/UI.java [hunk 4]
```diff
@@ -0,0 +1,137 @@
...
+    }
+
+    /**
+     * Says goodbye to the user.
+     */
+    public void bye() {
+        System.out.println("\tBye. zzz FINALLY~~" + " " + EMOJI);
+        reader.close();
+    }
+}
```
Comment (highorbit25) [inline src/main/java/chatbot/ui/UI.java:152]: Generally well-written code with no major code quality issues! Keep up the good work!

File: src/main/java/chatbot/tasks/TaskList.java (hunk 3 above)
Comment (zlimez) [reply src/main/java/chatbot/tasks/TaskList.java]: Yap haha just included checkstyle and found that out

File: src/main/java/chatbot/storage/Storage.java (hunk 1 above)
Comment (zlimez) [reply src/main/java/chatbot/storage/Storage.java]: Thanks

File: src/main/java/chatbot/parser/Parser.java [hunk 5]
```diff
@@ -0,0 +1,196 @@
...
+     * The method parses the argument segment of the user input
+     * to look for the task name and date of a time sensitive task.
+     *
+     * @param type The type of time sensitive task.
+     * @param args The argument to be parsed.
+     * @return An array containing the name and date of the task.
+     * @throws DukeException Raised if the argument does not contain
+     *      a task name and a date.
+     */
+    public static String[] parseTimedTask(String type, String args) throws DukeException {
```
Comment (richavm14) [inline src/main/java/chatbot/parser/Parser.java:174]: I like the use of Egyptian style braces and whitespace

File: src/main/java/chatbot/parser/Parser.java [hunk 6]
```diff
@@ -0,0 +1,196 @@
...
+import chatbot.commands.Unmark;
+import chatbot.exceptions.DukeException;
+import chatbot.tasks.Deadline;
+import chatbot.tasks.Event;
+
+/**
+ * The class is responsible for parsing user input and determining the consequent action.
+ */
+public class Parser {
+    public static final Pattern EVENT_FORMAT = Pattern.compile("(?<taskName>.+)\\s+/at\\s+(?<date>\\S+)");
```
Comment (richavm14) [inline src/main/java/chatbot/parser/Parser.java:33]: I like that you followed naming convention

File: src/main/java/chatbot/commands/Unmark.java [hunk 7]
```diff
@@ -0,0 +1,44 @@
+package chatbot.commands;
+
+import chatbot.exceptions.DukeException;
+import chatbot.tasks.Task;
+import chatbot.tasks.TaskList;
+import chatbot.ui.Response;
+import chatbot.ui.UI;
```
Comment (richavm14) [inline src/main/java/chatbot/commands/Unmark.java:7]: I like that you kept the order of imports consistent!

File: src/main/java/chatbot/tasks/Task.java [hunk 8]
```diff
@@ -0,0 +1,48 @@
+package chatbot.tasks;
+
+/**
+ * The class represents generic task like that
+ * of the real world with completion status and task name
+ * associated.
+ */
+public abstract class Task {
+    private boolean isComplete = false;
```
Comment (richavm14) [inline src/main/java/chatbot/tasks/Task.java:9]: Boolean variable is named to sound like boolean 👍

**Review by richavm14 — COMMENTED** [review]
You have followed coding standards. Good work!

**Pull request closed without merging**